source/common/http/http1/codec_impl.cc

```diff
@@ -514,6 +516,15 @@ int ConnectionImpl::onHeadersCompleteBase() {
       ENVOY_CONN_LOG(trace, "codec entering upgrade mode.", connection_);
       handling_upgrade_ = true;
     }
+  } else if (connection_header_sanitization_ && current_header_map_->Connection()) {
+    // If we fail to sanitize the request, return a 400 to the client
+    if (!Utility::sanitizeConnectionHeader(*current_header_map_)) {
+      absl::string_view header_value = current_header_map_->Connection()->value().getStringView();
+      ENVOY_CONN_LOG(debug, "Too many nominated headers in Connection: {}", connection_,
+                     header_value);
+      error_code_ = Http::Code::BadRequest;
+      sendProtocolError();
```

Comment:
I'd encourage an end to end test of this one - you should be able to crib off of IntegrationTest, NoHost or BadPath, and just have one for TooManyHeadersInConnection or something

Reply:
I believe I added a test that covers this case.  TestTooManyNominatedHeaders sends 12 nominated headers and expects a 400 back.

I could be misunderstanding your comment.